internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestBatch.java

```diff
@@ -650,6 +653,41 @@ public void testKafkaInputAAStore() throws Exception {
     testRepush(storeName, validator);
   }
 
+  @Test(timeOut = TEST_TIMEOUT)
+  public void testKafkaInputBatchJobSnapshots() throws Exception {
+
+    VPJValidator validator = (avroClient, vsonClient, metricsRepository) -> {
+      for (int i = 1; i <= 100; i++) {
+        Assert.assertEquals(avroClient.get(Integer.toString(i)).get().toString(), "test_name_" + i);
+      }
+    };
+
+    String storeName = testBatchStore(
+        inputDir -> new KeyAndValueSchemas(writeSimpleAvroFileWithStringToStringSchema(inputDir)),
+        properties -> {},
+        validator);
+
+    // Re-push with Kafka Input
+    testRepush(storeName, validator);
+
+    Path folderPath1 = Paths.get(BASE_DATA_PATH_1);
```

Comment:
Can we move this line up to 669 and verify there's snapshots created for version 1 but there's nothing for `BASE_DATA_PATH_2`? Then after line 674, we verify BASE_DATA_PATH_2 also has snapshots created so it validates the full lifecycle of regular push and repush.

Reply:
hmm. It looks like `BASE_DATA_PATH_1` and `BASE_DATA_PATH_2` belong to different venice server ([1](https://github.com/linkedin/venice/blob/main/internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestBatchForIngestionIsolation.java#L32) and [2](https://github.com/linkedin/venice/blob/main/internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestBatchForIngestionIsolation.java#L34)).  

I think you mean to test the following ? - 
1. testBatchStore (first push) : make sure we have v1 w/ snapshots but not any other versions 
2. testRepush (consecutive pushes) : make sure we have v2 w/ snapshots

i can cover this test case but pl lmk if thats not what you mean

Reply:
Yes, that's what I was asking for. I thought these 2 paths are for different versions but I could be wrong. If you can do that, it'd be great.